joblib/test/test_memory.py

```diff
@@ -410,33 +387,38 @@ def test_persistence(tmpdir):
     assert memory.cachedir == memory2.cachedir
 
     # Smoke test that pickling a memory with cachedir=None works
-    memory = Memory(cachedir=None, verbose=0)
-    pickle.loads(pickle.dumps(memory))
+    pickle.loads(pickle.dumps(no_memoize_memory))
     g = memory.cache(f)
     gp = pickle.loads(pickle.dumps(g))
     gp(1)
 
 
-def test_call_and_shelve(tmpdir):
+@fixture(scope='function', params=range(4))
```

Comment:
@lesteve: Well another obvious solution, till we dont get a better feature in pytest is to manually make a tempdir and remove it at such places. I am quite fond of `tmpdir` fixture because we don't need to care about isolation, creation and deletion and leakages at any point. Although if it is suitable, we can directly use the `@parametrize` decorator  by not using `tmpdir` _for this test only. (I can add a _TODO_ comment there at the best. Or is the current design fine ?)